doc/serial-timeout-troubleshoot.md

```diff
@@ -0,0 +1,80 @@
+
+Serial ports might be [UARTs](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52840.ps/uart.html?cp=2_0_0_5_32)
+on the devkits connected to a RS-232 port on a computer,
+or they might be emulated through USB CDC ACM (also known as "Virtual COM port",
+"serial over USB" or other names). The later happens when using 
+[USB ACM CDC on the IMCU](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52/dita/nrf52/development/preview_dev_kit/vir_com_port.html?cp=2_1_5_4_1),
+or when using [USB functionality from the Nordic SDK](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/lib_usbd_class_cdc.html?cp=4_0_0_3_51_8_3)
+on SoCs that implement USB (e.g. the nRF52840).
+
+While serial ports are ubiquitous, serial port transport lacks any protocol
+metadata or capabilities for request-response higher-level protocols. Combined
+with the fact that modern desktop operating systems still allow for legacy
+serial ports, this can trigger timeouts.
+
+This document collects some known scenarios and workarounds for these timeout 
+errors.
+
+# Scenarios where these errors happen
+
+
+## DevKit/dongle not running bootloader
+
+When using a nRF52840 devkit/dongle and performing a DFU operation with 
+nRF Connect Desktop, a `Timeout while reading from serial transport.` error shows up.
+
+When using a nRF52840 devkit/dongle and performing a DFU operation with
+[nrfutil](https://github.com/NordicSemiconductor/pc-nrfutil/), a 
+`TypeError: 'NoneType' object is not iterable` error shows up.
+
+When performing a DFU operation, nrfutil *assumes* that the serial port
+specified is a devkit/dongle already in bootloader mode. nRF Connect Desktop (and
+the underlying [`nrf-device-setup-js`](https://github.com/NordicSemiconductor/nrf-device-setup-js) library)
+will try to perform a [DFU trigger](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/lib_dfu_trigger_usb.html) 
+first, but the DFU trigger might fail due to other reasons.
+
+In either case, the firmware in the nRF SoC is receiving DFU data through the
+serial port, but is not understanding it, and it's not replying to it.
+
+#### Workaround
+
+Ensure that the devkit/dongle is in bootloader mode.
+
+- For nRF52840 dongles:
+    - Unplug it from its USB port
+    - Hold down the reset button.
+    - Plug the dongle into a USB port. The red LED should be pulsing, indicating the nRF device is in DFU bootloader mode.
+    - Stop holding down the reset button.
+    - Perform the DFU operation again.
+
+- For nRF52840 development kits:
+    - Unplug the devkit's USB connector marked "nRF USB", and connect a USB cable to the USB connector for the [Interface MCU](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52/dita/nrf52/development/nrf52840_pdk/if_mcu.html)
+    - Use nRF Connect Destkop Programmer to program it with bootloader firmware known to work.
+    - Perform the DFU operation again.
```

Comment:
I think we could avoid mistakes if we insert a step here e.g. _Unplug the devkit from the IMCU and plug it in via "nRF USB"_.